tests/types/index.ts

```diff
@@ -81,7 +81,61 @@ async function invokeSomeApis(service: cpl.Service) {
   let d: string = ''
   responseAsStream.on('data', data => { d += data.toString() })
   responseAsStream.on('end', () => console.log(d))
+}
+
+async function invokeSomeApis(service: cpl.HttpClient) {
+  service.get('/path')
+  service.get('/path', { query: { query: 'params'} })
+  service.get('/path', { returnAs: 'JSON', query: { query: 'params' } })
+  service.get('/path', { returnAs: 'BUFFER', query: { query: 'params' } })
+  service.get('/path', { returnAs: 'STREAM', query: { query: 'params' } })
+
+  service.post('/path', {my: 'body'})
+  service.post('/path', 'My body as string')
+  service.post('/path', Buffer.from('My body as buffer'))
+  service.post('/path', new stream.Readable())
+  service.post('/path', {my: 'body'}, { query: { query: 'params'} })
+  service.post('/path', {my: 'body'}, { returnAs: 'STREAM', query: { query: 'params'} })
+
+  service.put('/path', {my: 'body'})
+  service.put('/path', 'My body as string')
+  service.put('/path', Buffer.from('My body as buffer'))
+  service.put('/path', new stream.Readable())
+  service.put('/path', {my: 'body'}, { query: { query: 'params'} })
+  service.put('/path', {my: 'body'}, { returnAs: 'STREAM', query: { query: 'params'} })
 
+  service.patch('/path', {my: 'body'})
+  service.patch('/path', 'My body as string')
+  service.patch('/path', Buffer.from('My body as buffer'))
+  service.patch('/path', new stream.Readable())
+  service.patch('/path', {my: 'body'}, {query: { query: 'params'}})
+  service.patch('/path', {my: 'body'}, { returnAs: 'STREAM', query: { query: 'params'} })
+
+  service.delete('/path', {my: 'body'})
+  service.delete('/path', 'My body as string')
+  service.delete('/path', Buffer.from('My body as buffer'))
+  service.delete('/path', new stream.Readable())
+  service.delete('/path', {my: 'body'}, { query: { query: 'params'} })
+  service.delete('/path', {my: 'body'}, { returnAs: 'STREAM', query: { query: 'params'} })
+
+  interface Res {
+    foo: string
+  }
+  const response = await service.get<cpl.JSONResponse<Res>>('/path')
+  console.log(response.statusCode)
+  console.log(response.headers)
+  console.log(response.payload.foo)
+
+  const responseAsBuffer = await service.get<cpl.BufferResponse>('/path')
+  console.log(responseAsBuffer.statusCode)
```

Comment:
we need these console logs?

Reply:
At the moment, tests on ts definition are made in this way. It should be something to change, using for example [tsd](https://github.com/SamVerschueren/tsd)